From 32d2aa2cea22bb38fc3eeded1f6682daad097908 Mon Sep 17 00:00:00 2001
From: Gauvain Pocentek <gauvainpocentek@gmail.com>
Date: Tue, 13 Aug 2019 20:11:21 +0200
Subject: [PATCH] Fix argument parsing for region_name

The region_name attribute for fence_compute and fence_evacuate is not a boolean
and should expect a string argument.
---
 agents/compute/fence_compute.py        | 2 +-
 agents/evacuate/fence_evacuate.py      | 2 +-
 tests/data/metadata/fence_compute.xml  | 2 +-
 tests/data/metadata/fence_evacuate.xml | 2 +-
 4 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/agents/compute/fence_compute.py b/agents/compute/fence_compute.py
index 0e6c8044..a94bdc46 100644
--- a/agents/compute/fence_compute.py
+++ b/agents/compute/fence_compute.py
@@ -381,7 +381,7 @@ def define_new_opts():
 		"order": 1,
 	}
 	all_opt["region_name"] = {
-		"getopt" : "",
+		"getopt" : ":",
 		"longopt" : "region-name",
 		"help" : "--region-name=[region]         Region Name",
 		"required" : "0",
diff --git a/agents/evacuate/fence_evacuate.py b/agents/evacuate/fence_evacuate.py
index 2b9af273..76e78b87 100644
--- a/agents/evacuate/fence_evacuate.py
+++ b/agents/evacuate/fence_evacuate.py
@@ -318,7 +318,7 @@ def define_new_opts():
 		"order": 1,
 	}
 	all_opt["region_name"] = {
-		"getopt" : "",
+		"getopt" : ":",
 		"longopt" : "region-name",
 		"help" : "--region-name=[region]                 Region Name",
 		"required" : "0",
diff --git a/tests/data/metadata/fence_compute.xml b/tests/data/metadata/fence_compute.xml
index 1dcbfc54..abc276ca 100644
--- a/tests/data/metadata/fence_compute.xml
+++ b/tests/data/metadata/fence_compute.xml
@@ -55,7 +55,7 @@
 	</parameter>
 	<parameter name="region_name" unique="0" required="0">
 		<getopt mixed="--region-name=[region]" />
-		<content type="boolean" default=""  />
+		<content type="string" default=""  />
 		<shortdesc lang="en">Region Name</shortdesc>
 	</parameter>
 	<parameter name="tenant_name" unique="0" required="0">
diff --git a/tests/data/metadata/fence_evacuate.xml b/tests/data/metadata/fence_evacuate.xml
index 4f1f6a58..7decb05c 100644
--- a/tests/data/metadata/fence_evacuate.xml
+++ b/tests/data/metadata/fence_evacuate.xml
@@ -55,7 +55,7 @@
 	</parameter>
 	<parameter name="region_name" unique="0" required="0">
 		<getopt mixed="--region-name=[region]" />
-		<content type="boolean" default=""  />
+		<content type="string" default=""  />
 		<shortdesc lang="en">Region Name</shortdesc>
 	</parameter>
 	<parameter name="tenant_name" unique="0" required="0">
